smart-contract/src/contract.rs

```diff
@@ -223,28 +224,32 @@ fn try_vote(
         return Err(contract_err("member has already voted"));
     }
 
-    // Esure member joined before the proposal was created and determine weight.
-    let weight = if let Some(m) = member {
+    if info.sender == state.admin && proposal.admin_vote != None {
```

Comment:
is there a `proposal.admin_vote.is_some()`? I believe that would be more idiomatic.

Reply:
ah thanks